# algorithm param

# trajecotry total/fixing length
trajectory_length: 200
forward_fixing_distance: 10.0
backward_fixing_distance: 5.0

# clearance(distance) when generating trajectory
min_clearance_from_road: 0.2
min_clearance_from_object: 1.2
# clearance for unique points
clearance_for_straight_line_: 0.05
clearance_for_joint_: 3.2
clearance_for_only_smoothing: 0.1
clearance_from_object_for_straight: 10.0

# avoiding objects less than this velocity
max_avoiding_objects_velocity_ms: 1.1

# solving quadratic programming
qp_max_iteration: 10000
qp_eps_abs: 1.0e-8
qp_eps_rel: 1.0e-11
qp_eps_abs_for_extending: 1.0e-6
qp_eps_rel_for_extending: 1.0e-8



# constrain space
min_lat_constrain_space: 0.1
min_lon_constrain_space: 0.1
max_x_constrain_search_range: 0.4
coef_x_cosntrain_search_resolution: 1.0
coef_y_cosntrain_search_resolution: 0.5
keep_space_shape_x: 2.0
keep_space_shape_y: 2.0

is_publishing_clearance_map: false
enable_avoidance: true
num_sampling_points: 100
num_joint_buffer_points: 3
num_offset_for_begin_idx: 2
num_fix_points_for_extending: 2
delta_arc_length_for_optimization: 1.0
delta_arc_length_for_trajectory: 0.1
delta_yaw_threshold_for_closest_point: 1.046  #M_PI/3.0
delta_yaw_threshold_for_straight: 0.02
# end algoritm param

# replanning & trimming trajectory param outside algorithm
min_delta_dist_for_replan: 10.0
min_delta_time_sec_for_replan: 1.0
max_dist_for_extending_end_point: 5.0
distance_for_path_shape_chagne_detection: 2.0
